src/main/java/sirius/biz/storage/layer3/JobFileQuickActionProvider.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.storage.layer3;
+
+import sirius.biz.jobs.JobFactory;
+import sirius.biz.tycho.QuickAction;
+import sirius.kernel.async.CallContext;
+import sirius.kernel.di.std.PriorityParts;
+import sirius.kernel.di.std.Register;
+import sirius.web.http.WebContext;
+import sirius.web.security.UserContext;
+import sirius.web.security.UserInfo;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * Provides job based quick actions for files. (Short-cut to start a job for a given file).
+ */
+@Register
+public class JobFileQuickActionProvider implements FileQuickActionProvider {
+
+    /**
+     * Defines the permission required to execute jobs.
+     */
+    public static final String PERMISSION_EXECUTE_JOBS = "permission-execute-jobs";
+
+    @PriorityParts(JobFactory.class)
+    private List<JobFactory> factories;
+
+    /**
+     * Returns a stream of {@link JobFactory jobs} available for the current user.
+     *
+     * @return a stream of jobs which are available to the current user and match the given search query
+     */
+    public Stream<JobFactory> getAvailableJobs() {
+        UserInfo currentUser = UserContext.getCurrentUser();
+        if (!currentUser.hasPermission(PERMISSION_EXECUTE_JOBS)) {
+            return Stream.empty();
+        }
+
+        Stream<JobFactory> stream = factories.stream().filter(JobFactory::isAccessibleToCurrentUser);
```

Comment:
Do we need the permission check above? Doesn't isAccessibleToCurrentUser don't already do that? Besides, BasicJobFactory.generatePresetUrl already validates that only interactive jobs available to the current user returns something here.

Reply:
I dont know. I copied it from the old code (which did exactly that), so I assumed it is necessary

Reply:
OK, does not harm to keep it. Would be cool to know if is obsolete, would expect someone cannot start jobs at all shall  get isAccessibleToCurrentUser = false, and if not, perhaps that check could be moved over there?
See what others say...